c2rust-analyze/src/util.rs

```diff
@@ -330,20 +334,24 @@ pub fn are_transmutable<'tcx>(a: Ty<'tcx>, b: Ty<'tcx>) -> bool {
         }
     };
 
-    // only check for transmutable ints so far
-    a == b || transmutable_ints
+    let one_way_transmutable = || match from_ty.kind() {
+        &ty::Array(from_ty, _) | &ty::Slice(from_ty) => is_transmutable_to(from_ty, to_ty),
+        _ => false,
+    };
+
+    from_ty == to_ty || transmutable_ints() || one_way_transmutable()
```

Comment:
are these function calls here for laziness?

Reply:
Yeah.  I want to put all the main operations in one line for clarity, but run them in the specific order I want.